### ISSUE TITLE
Obsidian Dynamic Data Block Undesirable/Unreadable View with Large Column Count

### ISSUE THREAD

**Issue by mikedotmundy** [SparkDevNetwork/Rock#5959]
### Description

Using the new Obsidian Dynamic Data Block, if you have a query that returns a large number of columns, the view is undesirable and most of the data unreadable. 

Consider this example: You have a page with a Dynamic Data Block that shows Pages with long load times.
Query:
```
SELECT *
FROM [Page]
WHERE [MedianPageLoadTimeDurationSeconds] > 1
ORDER BY [MedianPageLoadTimeDurationSeconds] DESC
```

Here is what it looks like:
![Screenshot 2024-07-17 at 10 41 30](https://github.com/user-attachments/assets/c064d1e0-d50b-4bc7-94e6-739c3f39e3bb)

Here is a screen recording showing what happens if you scroll horizontally:
https://github.com/user-attachments/assets/25a78c8d-efca-4766-a3df-4d31db61bac7

2 issues:
1. The columns are incredibly skinny, making the data unreadable
2. The columns go off-grid to the right of the page, making their data unreadable even when scrolling horizontally. 

### Actual Behavior

Data unreadable when large number of columns are returned.

### Expected Behavior

Like the Dynamic Data Webforms block, you should be able to return large number of columns. The columns are sized appropriately and you can scroll horizontally inside the block, not off the page.

### Steps to Reproduce

- Add the code above to an Obsidian Dynamic Data Block
- See results as shown above.

### Issue Confirmation

- [X] Perform a search on the [Github Issues](https://github.com/issues?q=is%3Aissue+user%3ASparkDevNetwork+-repo%3ARock) to see if your bug or enhancement is already reported.
- [X] Reproduced the problem on a fresh install or on the [demo site](https://rock.rocksolidchurchdemo.com/).

### Rock Version

v16.6, v17.0

### Client Culture Setting

en-US

**Comment by stanalyst** [comment]
I've experience some additional related issues with the Obsidian Dynamic Data block.

![ObsidianDynData-1](https://github.com/user-attachments/assets/66fd28ed-0cef-4d9f-8265-ee7cea2c97c8)
![ObsidianDynData-2](https://github.com/user-attachments/assets/dfcfd1ee-66b4-4a85-838a-9740c3364767)
![ObsidianDynData-3](https://github.com/user-attachments/assets/6aa25acb-5425-4f00-b895-9659bc49bf4b)
![ObsidianDynData-4](https://github.com/user-attachments/assets/256104d9-b634-4b59-8e99-ca0cc826dd53)
![ObsidianDynData-5](https://github.com/user-attachments/assets/5c89864e-7bfe-4aa5-aa91-0cbcdbcfcf0f)
![ObsidianDynData-6](https://github.com/user-attachments/assets/cefa38af-5bfc-4b26-a8b4-57d7511dd45d)

**Comment by kylehenning** [comment]
This issue has been resolved with the recent commit e7e52e7.
